hw/misc/hermes.c

```diff
@@ -960,12 +995,35 @@ static void pci_hermes_realize(PCIDevice *pdev, Error **errp)
             PCI_BASE_ADDRESS_MEM_TYPE_64, &hermes->bar4_mem_reg);
 
     hermes_init_msix(hermes, errp);
+
+    hermes->stopping = false;
+    qemu_mutex_init(&hermes->bar2.h2c.dma_mutex);
```

Comment:
Might be worth having a function to init and destroy the dir_regs so we don't have to double up on each init and have long hermes->bar2.... prefixes on every variable.